src/SFA.DAS.ApprenticeCommitments.Jobs.Functions/EventHandlers/DomainEvents/ApprenticeEmailAddressChangedHandler.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.Extensions.Logging;
+using NServiceBus;
+using SFA.DAS.ApprenticeAccounts.Messages.Events;
+using SFA.DAS.ApprenticeCommitments.Jobs.Api;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeCommitments.Jobs.Functions.EventHandlers.DomainEvents
+{
+    public class ApprenticeEmailAddressChangedHandler : IHandleMessages<ApprenticeEmailAddressChanged>
+    {
+        private readonly IEcsApi _api;
+        private readonly ILogger<ApprenticeshipChangedEventHandler> _logger;
+
+        public ApprenticeEmailAddressChangedHandler(IEcsApi api, ILogger<ApprenticeshipChangedEventHandler> logger)
+        {
+            _api = api;
+            _logger = logger;
+        }
+
+        public async Task Handle(ApprenticeEmailAddressChanged message, IMessageHandlerContext context)
+        {
+            _logger.LogInformation("Apprentice {ApprenticeId} changed their email address", message.ApprenticeId);
+            await _api.UpdateApprentice(message.ApprenticeId, new JsonPatchDocument<Apprentice>());
```

Comment:
Don't think this is a good pattern.  It's not passing the new email address to the patch document event, I know the value is not needed in the API, but surely we only want to invoke the updates to commitments when the email address has been changed. If this patch endpoint contained updates for another field (in the future). We would not want to invoke the updates to commitments.

Reply:
Yes, I agree.  It boils down to the fact that the apprentice is no longer a concern of the API, and so trying to shoehorn a change-of-apprentice-email responsibility into the domain ends up with a strange looking API.

The obvious (once I thought of it!) solution is to make jobs responsible for publishing the `ApprenticeshipEmailAddressChangedEvent` instead - no weird API needed.